src/Microsoft.ML.LightGbm/LightGbmMulticlassTrainer.cs

```diff
@@ -34,9 +40,61 @@ public sealed class LightGbmMulticlassClassificationTrainer : LightGbmTrainerBas
         private int _tlcNumClass;
         private protected override PredictionKind PredictionKind => PredictionKind.MulticlassClassification;
 
+        public sealed class Options : OptionsBase
+        {
+            public enum EvaluateMetricType
+            {
+                None,
+                Default,
+                Error,
+                LogLoss,
+            }
+
+            /// <summary>
+            /// Whether to use softmax loss.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Use softmax loss for the multi classification.")]
+            [TlcModule.SweepableDiscreteParam("UseSoftmax", new object[] { true, false })]
+            public bool? UseSoftmax;
+
+            /// <summary>
+            /// Parameter for the sigmoid function.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Parameter for the sigmoid function.", ShortName = "sigmoid")]
+            [TGUI(Label = "Sigmoid", SuggestedSweeps = "0.5,1")]
+            public double Sigmoid = 0.5;
+
+            /// <summary>
+            /// Determines what evaluation metric to use.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce,
+                HelpText = "Evaluation metrics.",
+                ShortName = "em")]
+            public EvaluateMetricType EvaluationMetric = EvaluateMetricType.Error;
+
+            static Options()
+            {
+                NameMapping.Add(nameof(EvaluateMetricType), "metric");
+                NameMapping.Add(nameof(EvaluateMetricType.Error), "multi_error");
+                NameMapping.Add(nameof(EvaluateMetricType.LogLoss), "multi_logloss");
+            }
+
```

Comment:
This might not work with new Options().

Reply:
Since this is static, it will be created before our main function is called.

---
In reply to: [266672901](https://github.com/dotnet/machinelearning/pull/2948#discussion_r266672901) [](ancestors = 266672901)